From: Manoj Srivastava <srivasta@debian.org>
Date: Thu, 3 Apr 2008 02:04:02 -0500
Subject: Fix directory search for link libraries

Bug fix: "make: Directory Search for Link Libraries does not look in
/usr/local/lib", thanks to Daniel Barlow. In the Debian binaries,
PREFIX is /usr and thus this searches /lib, /usr/lib and /usr/lib
again and therefore misses any libraries that are not packaged and
were installed by the site admin. The ideal behaviour would be to
have the search path set by a Makefile variable (other than the VPATH
blunt object) but even absent that, it would be more useful if it
looked in /usr/local/lib even though make itself hasn't been installed
in the /usr/local tree.

Last-Update: 2023-02-12
---
 src/remake.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/src/remake.c
+++ b/src/remake.c
@@ -1701,6 +1701,18 @@
 #define LIBDIR "."
 #endif
       LIBDIR,                   /* Defined by configuration.  */
+#ifndef _AMIGA
+/*
+ * In the Debian binaries, PREFIX is /usr and thus this searches /lib,
+ * /usr/lib and /usr/lib again and therefore misses any libraries that
+ * are not packaged and were installed by the site admin.  The ideal
+ * behaviour would be to have the search path set by a Makefile
+ * variable (other than the VPATH blunt object) but even absent that,
+ * it would be more useful if it looked in /usr/local/lib even though
+ * make itself hasn't been installed in the /usr/local tree -- manoj
+ */
+      "/usr/local/lib",
+#endif
       0
     };
 
